src/collection.ts

```diff
@@ -240,6 +238,14 @@ export class Collection implements OperationParent {
     return this.s.bsonOptions;
   }
 
+  /**
+   * The current topology of the collection.
+   * @internal
+   */
+  getTopology(): Topology | undefined {
```

Comment:
a-ha, when we were discussing this the other day I meant something a little different than this - I see how the suggestion would be interpreted this way, sorry for the confusion. What I think we should really try to do as much as possible is reduce the public api impact of this change. Rather than providing a `getTopology` method (as opposed to a `topology` property), I was suggesting that we provide some sort of completely internal `getTopology` method that _accepts_ something that provides a topology:

```
function getTopology(provider: MongoClient | Db | Collection): Topology {
  if (provider.s.topology) return provider.s.topology;
  if ....
}
```

A stricter approach might be to introduce a `TopologyProvider` interface, maybe in `topology.ts`:
```
const kTopology = Symbol('topology');
export interface TopologyProvider {
  [kTopology]: Topology
}

export getTopology(provider: TopologyProvider): Topology {
  return provider[kTopology];
}
```
If that approach works, then you have made `topology` visible and not accessible outside of the library (private)